### PR TITLE
[5.1] whereIn collection support

### PR DESCRIPTION
**Pull request by jbrooksuk**
Replaces #9014 :)

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
:-1:

**Comment by GrahamCampbell** [comment]
Better fix would be to change `$this->addBinding($values, 'where');` to `$this->addBinding($values->toArray(), 'where');`

**Comment by jbrooksuk** [comment]
@GrahamCampbell is your fix right though? `$values` will be an array by default, so there is no `toArray` method on it?

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -748,6 +748,8 @@ public function whereIn($column, $values, $boolean = 'and', $not = false)
         // execute those Closures, then we can re-construct the entire sub-selects.
         if ($values instanceof Closure) {
             return $this->whereInSub($column, $values, $boolean, $not);
+        } elseif ($values instanceof Collection) {
+            $values = $values->toArray();
```
Comment (crynobone) [inline src/Illuminate/Database/Query/Builder.php]: I would prefer to check for `Illuminate\Contracts\Support\Arrayable`.

**Comment by GrahamCampbell** [comment]
> is your fix right though? $values will be an array by default, so there is no toArray method on it?

Maybe not, lol. Still, get rid of the elseif, because it's confusing, and leave blank line, then start a new iff statament. @crynobone is also correct about the Arrayable thing.

**Comment by jbrooksuk** [comment]
Ok cool, will fix them :)

**Comment by jbrooksuk** [comment]
@GrahamCampbell all done!

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -5,6 +5,7 @@
 use Closure;
 use BadMethodCallException;
 use InvalidArgumentException;
+use Illuminate\Contracts\Support\Arrayable;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: Please sort imports by length.

**Comment by GrahamCampbell** [comment]
:+1:

**Pull request merged**